### DISCUSSION TITLE
Simulation not working on macOS Ventura 13.4

### DISCUSSION THREAD

**Discussion by mihaylichenko** [zepp-health/zeppos-forum#200] [Q&A] [answered]
Hello! When I try to run the simulation, I get errors in the console:
`[firmware_simulator]: chmod: Unable to change file mode on qemu_mac/qemu-system-arm: Operation not permitted
[firmware_simulator]: chmod: Unable to change file mode on qemu_mac/qemu-ifup: Operation not permitted
[firmware_simulator]: chmod: Unable to change file mode on qemu_mac/qemu-ifdown: Operation not permitted
[firmware_simulator]: ./start_qemu.sh: line 141: ./qemu_mac/qemu-system-arm: Permission denied`

The virtual network card is installed, the command `ls /dev` displays a list tap0 - tap1

System Options
MacBook Pro  Intel Core i7

**Comment by MyoungXUE** [comment, marked as answer]
Hello, from your description, it looks like a lack of permission.

Please refer to the Zepp OS official documentation simulator the section on https://docs.zepp.com/docs/guides/tools/simulator/setup/#install-and-launch-the-simulator-1.

This command is required to start the simulator.

```sh
cd /Applications/simulator.app/Contents/MacOS && sudo -s ./simulator
```

**Reply by mihaylichenko** [reply]
You were right, thanks!
MaсOS did not give me any notifications about this (

**Comment by LightningK0ala** [comment]
I had to disable SIP on a Mac M2 to get past this error, but now it just says `[firmware_simulator]: simulator opening...` forever. I've tried this on the Falcon v1.0.0 and the GTR mini v2.0.0 simulators. Also, I am trying to run a GTR mini project and a message shows up in the terminal `inject stop because application is inconsistent with the watch simulator model` when I click the app I'm trying to run.

**Reply by MyoungXUE** [reply]
@LightningK0ala Hello!

Every device with Zepp OS, including simulators, has a deviceSource attribute, you can refer to https://docs.zepp.com/docs/reference/related-resources/device-list/ for it.

The platforms field in the `app.json` configuration of the Mini Program specifies the device on which the Mini Program is running, see https://docs.zepp.com/docs/reference/app-json/#platforms-object.

Only if the platforms field in `app.json` matches the deviceSource of the corresponding simulator, the app will be installed correctly on the simulator.
